src/devices/Display/FontHelper14.cs

```diff
@@ -0,0 +1,225 @@
+        /// </summary>
+        /// <param name="digit">hexadecimal digit (0..F)</param>
+        /// <returns>corresponding font</returns>
+        public static Font14 GetHexDigit(byte digit) => s_hexDigits[digit & HexadecimalMask];
+
+        /// <summary>
+        /// Converts a span of bytes to their corresponding hexadecimal digits' font representation
+        /// </summary>
+        /// <param name="digits">list of hexadecimal digits (will be converted in place!)</param>
+        public static void ConvertHexDigits(Span<byte> digits)
+        {
+            for (int i = 0, l = digits.Length; i < l; i++)
+            {
+                digits[i] = (byte)GetHexDigit(digits[i]);
+            }
+        }
+
+        /// <summary>
+        /// Converts a span of bytes to their corresponding hexadecimal digits' font representation
+        /// </summary>
+        /// <param name="digits">list of hexadecimal digits (will be converted in place!)</param>
+        /// <returns>list of corresponding digit fonts</returns>
+        public static Font14[] GetHexDigits(ReadOnlySpan<byte> digits)
+        {
+            var fonts = new Font14[digits.Length];
+            if (digits.Length > 0)
+            {
+                for (int i = 0, l = fonts.Length; i < l; i++)
+                {
+                    fonts[i] = GetHexDigit(digits[i]);
+                }
+            }
+
+            return fonts;
+        }
+
+        /// <summary>
+        /// Convert character value to corresponding font segments
+        /// </summary>
+        /// <param name="value">input character</param>
+        /// <returns>corresponding font bits</returns>
+        public static Font14 GetCharacter(char value) => value switch
+        {
+            _ when value >= 'A' && value <= 'Z' => s_upperCaseLetters[value - 'A'],
+            _ when value >= 'a' && value <= 'z' => s_lowerCaseLetters[value - 'a'],
+            _ when value >= '0' && value <= '9' => s_hexDigits[value - '0'],
+            '-' => Font14.Symbol_Minus,
+            '=' => Font14.Symbol_Equals,
+            '_' => Font14.Symbol_Underscore,
+            '|' => Font14.Symbol_Pipe,
+            '°' => Font14.Symbol_Degree,
+            '[' => Font14.Symbol_LeftSquareBracket,
+            ']' => Font14.Symbol_RightSquareBracket,
+            '.' => Font14.Symbol_FullStop,
+            _ => Font14.Whitespace
+        };
+
+        /// <summary>
+        /// Convert a string of characters to corresponding fonts
+        /// </summary>
+        /// <param name="input">input string</param>
+        /// <param name="output">list of corresponding character fonts</param>
+        public static void ConvertString(ReadOnlySpan<char> input, Span<Font14> output)
+        {
+            /*if (input.Length != output.Length)
+            {
+                throw new InvalidOperationException($"{nameof(input)} and {nameof(output)} length must be the same");
+            }*/
```

Comment:
Why is this test commented out? It seems to be a valid check.

Reply:
It's not a valid test. In cases where decimal points are represented then their input chars (say '42.75') will be 5 but output will have the '2.' on a single segment. I could check for this of course, but my use case for this device didn't demand it. I played the priority card :-)

Reply:
Commented out code removed